src/library/export/coverartcopyworker.cpp

```diff
@@ -0,0 +1,98 @@
+#include "library/export/coverartcopyworker.h"
+
+#include <QDebug>
+#include <QFileInfo>
+#include <QMessageBox>
+
+#include "util/fileaccess.h"
+#include "util/imagefiledata.h"
+#include "util/safelywritablefile.h"
+
+void CoverArtCopyWorker::run() {
+    auto selectedCover = mixxx::FileAccess(mixxx::FileInfo(m_selectedCoverArtFilePath));
+    ImageFileData imageFileData = ImageFileData::fromFilePath(m_selectedCoverArtFilePath);
+    if (imageFileData.isNull()) {
+        // TODO(rryan): feedback
+        return;
+    }
+
+    m_coverInfo.type = CoverInfo::FILE;
+    m_coverInfo.source = CoverInfo::USER_SELECTED;
+    m_coverInfo.coverLocation = m_selectedCoverArtFilePath;
+    m_coverInfo.setImage(imageFileData);
+
+    if (QFileInfo(m_oldCoverArtFilePath).canonicalPath() ==
+            QFileInfo(m_selectedCoverArtFilePath).canonicalPath()) {
+        qDebug() << "Track and selected cover art are in the same path:"
+                 << QFileInfo(m_selectedCoverArtFilePath).canonicalPath()
+                 << "Cover art updated without copying";
+        emit updateCoverArt(m_coverInfo);
+        return;
```

Comment:
FileAccess already has e FileInfo member that should be used. QFileInfo is heavy, and makes use of caching.
It is the best to have only one instance per file.

Reply:
I think this block can be moved up a bit.

Reply:
> FileAccess already has e FileInfo member that should be used. QFileInfo is heavy, and makes use of caching. It is the best to have only one instance per file.

Done 👍 

Thank you for this thip.

> I think this block can be moved up a bit.

In order to update the cover art we need cover info, before this block we coverInfo comes. I couldn't understand where to move exactly. Could you please explain it a bit more?

Reply:
Oh, I missed that.